File: app-geekhome/src/main/java/eu/geekhome/rest/tags/TagsController.java
package eu.geekhome.rest.tags;

import eu.geekhome.rest.PluginsCoordinator;
import eu.geekhome.services.repository.Repository;
import eu.geekhome.services.repository.TagDto;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

@Path("tags")
public class TagsController {

    private final Repository _repository;

    @Inject
    public TagsController(PluginsCoordinator pluginsCoordinator) {
        _repository = pluginsCoordinator.getRepository();
    }

    @POST
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public long postTag(TagDto tagDto) {
        return _repository.saveTag(tagDto);
    }

    @PUT
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public void putTag(TagDto tagDto) {
        _repository.updateTag(tagDto);
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public List<TagDto> getAllTags() {
        return _repository.getAllTags();
    }

    @DELETE
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public void deleteTag(@PathParam("id") long id) {
        _repository.deleteTag(id);
    }
}

File: app-geekhome/src/main/java/eu/geekhome/rest/tags/TagsController.kt
package eu.geekhome.rest.tags

import eu.geekhome.data.Repository
import eu.geekhome.data.tags.TagDto
import javax.inject.Inject
import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("tags")
class TagsController @Inject constructor(private val repository: Repository) {

    @POST
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    fun postTag(tagDto: TagDto?): Long {
        return repository
            .saveTag(tagDto!!)
    }

    @PUT
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    fun putTag(tagDto: TagDto?) {
        repository
            .updateTag(tagDto!!)
    }

    @get:Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    @get:GET
    val allTags: List<TagDto>
        get() = repository
                    .getAllTags()

    @DELETE
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    fun deleteTag(@PathParam("id") id: Long) {
        repository
            .deleteTag(id)
    }
}